dbms/src/Common/Macros.cpp

```diff
@@ -57,16 +57,23 @@ String Macros::expand(const String & s, size_t level) const
 
         String macro_name = s.substr(begin, end - begin);
 
-        auto it = macros.find(macro_name);
-        if (it == macros.end())
-            throw Exception("No macro " + macro_name + " in config", ErrorCodes::SYNTAX_ERROR);
+        if (macro_name == "database")
```

Comment:
Explicitly defined "database" and "table" macros should be preferred over implicit values.